### PR TITLE
Updates to ethereumjs-config v2.0.0 (old)

### PR DESCRIPTION
**Pull request by evertonfraga**
This PR removes references to the old `ethereumjs-config` packages, replacing by the new ones. Configures all the new tooling.

This PR uses gitpkg urls to test monorepo package dependencies prior to their release. 

Remaining checks:
- [x] Documentation to use unpublished packages with Verdaccio.

## Fix lint results
- [x] Account
- [x] Block
- [x] Blockchain
- [x] Common
- [x] Ethash
- [x] Tx (blocked by #812)
- [x] Vm 

## Fix test runners
- [x] Account
- [x] Block
- [x] Blockchain
- [x] Common
- [x] Ethash 
- [x] Tx
- [x] Vm

## Other tasks
- [x] Fix coverage reporting for all packages
- [ ] Fix bash errors on script
- [ ] Point package references to NPM registry after Config release

### CODE REVIEW

**Comment by holgerd77** [comment]
Seems to be pretty "en vogue" in the team these days to have no PR below 40 files changed. 😂 🤣 😅

**Comment by ryanio** [comment]
haha oh yeah, careful with regenerating those docs too 😄

**Comment by evertonfraga** [comment]
@holgerd77 🤣 

I can move the docs to another PR, I think that's the most sensible thing to do.

**Comment by holgerd77** [comment]
@evertonfraga I don't care too much, think we will also get this reviewed with the docs, whatever is more convenient for you.

**Comment by evertonfraga** [comment]
After review, I'll release the `ethereumjs-config` 2.0.0, then update here on this branch and merge it!

File: packages/vm/.eslintrc.js [hunk 1]
```diff
@@ -0,0 +1,18 @@
...
+    '@typescript-eslint/no-use-before-define': 'off',
+    '@typescript-eslint/restrict-plus-operands': 'off',
+    'no-async-promise-executor': 'off',
+    'no-invalid-this': 'off',
+    'no-restricted-syntax': 'off',
+    'no-undef': 'off',
+    'no-unused-vars': 'off',
+    'prefer-const': 'off',
+    'sonarjs/no-duplicated-branches': 'off',
+  }
```
Comment (holgerd77) [inline packages/vm/.eslintrc.js:17]: Ah, so many exceptions for the VM necessary? 😄 

No problem of course on this round. Nevertheless interesting to state and see.

File: packages/vm/lib/evm/eei.ts [hunk 2]
```diff
@@ -349,14 +365,18 @@ export default class EEI {
   async _selfDestruct(toAddress: Buffer): Promise<void> {
     // only add to refund if this is the first selfdestruct for the address
     if (!this._result.selfdestruct[this._env.address.toString('hex')]) {
-      this.refundGas(new BN(this._common.param('gasPrices', 'selfdestructRefund')))
+      this.refundGas(
+        new BN(this._common.param('gasPrices', 'selfdestructRefund'))
+      )
```
Comment (holgerd77) [inline packages/vm/lib/evm/eei.ts:370]: I actually liked the `printWidth` setting being extended to 100 lines (from 800 or so) on the [old format package](https://github.com/ethereumjs/ethereumjs-config/pull/33/files#diff-5bc8394070975c0969eda8f8eb989c61L3), this gives a bit more space and there was a somewhat longer discussion around this during the introduction of config until we decided upon this.

This seems to not have made it, was this intentional? (at least I am assuming during re-formattings like the above, haven't concretely checked).

Not sure, should we keep / post-introduce this, any opinions @cgewecke @ryanio or @jochem-brouwer ?

@evertonfraga in doubt and if too much process interrupting merge here anyhow, we'll get this settled out later, eventually with some follow-up PR after discussion just with the linting updates.

File: packages/vm/package.json [hunk 3]
```diff
@@ -43,13 +41,13 @@
...
   "dependencies": {
-    "async-eventemitter": "^0.2.4",
-    "core-js-pure": "^3.0.1",
     "@ethereumjs/account": "^3.0.0",
     "@ethereumjs/block": "^3.0.0",
     "@ethereumjs/blockchain": "^4.0.3",
     "@ethereumjs/common": "^1.5.1",
     "@ethereumjs/tx": "^2.1.2",
+    "async-eventemitter": "^0.2.4",
+    "core-js-pure": "^3.0.1",
```
Comment (holgerd77) [inline packages/vm/package.json:50]: What are these two new dependencies?

**Review by holgerd77 — APPROVED** [review]
Nothing to hold off the merge, looks good, thanks Everton for the great work on this!

File: packages/account/tsconfig.browser.json [hunk 4]
```diff
@@ -0,0 +1,7 @@
+{
+  "extends": "@ethereumjs/config-typescript/tsconfig.browser.json",
+  "include": ["src/**/*.ts"],
+  "compilerOptions": {
+    "outDir": "./dist.browser"
+  }
+}
```
[the remark this answers is not in the document]
Comment (holgerd77) [reply packages/account/tsconfig.browser.json:7]: I would pass on this whole task to @cgewecke - so to eventually integrate and to check that these two new build targets are working as expected and eventually give side effects another thought - if both of you are ok with this.

File: packages/vm/.eslintrc.js (hunk 1 above)
Comment (evertonfraga) [reply packages/vm/.eslintrc.js:17]: yes! there were so many eslint complaints. I decided to add those as exceptions for now, and ship the config sooner, as it's working perfectly. 

We can tackle those in other PRs.

File: packages/account/.prettierrc [hunk 5]
```diff
@@ -0,0 +1,4 @@
+{
+  "semi": false,
+  "singleQuote": true
+}
```
[the remark this answers is not in the document]
Comment (evertonfraga) [reply packages/account/.prettierrc:4]: @holgerd77 So the linting process can occur independently on the CI and locally. 

I understand the repetition, and I can look into alternatives, but I'd rather play on the safer side first, where we can have more granular control over the `ethereumjs-config` packages

File: packages/account/tsconfig.browser.json (hunk 4 above)
[the remark this answers is not in the document]
Comment (evertonfraga) [reply packages/account/tsconfig.browser.json:7]: @holgerd77 I am OK with it, and can follow up with @cgewecke on that.

File: packages/vm/package.json (hunk 3 above)
Comment (evertonfraga) [reply packages/vm/package.json:50]: @holgerd77 they [were already there](https://github.com/ethereumjs/ethereumjs-vm/pull/886/files/7108dafe36bc9d4d00ad14abd2006b487a0cb2af#diff-76c14e78cd18d988b3bb6794290539fbL46). I have just moved them to the correct order (@, A-Z)

**Review by holgerd77 — DISMISSED** [review]
Reapproval after new commits.

**Comment by holgerd77** [comment]
@evertonfraga Thanks for the latest updates here, looks great 🙂 , let me know once you need a review!

**Comment by holgerd77** [comment]
Superseded by #913, will close.

**Pull request closed without merging**